python/ray/tune/examples/mnist_pytorch_lightning.py

```diff
@@ -176,8 +163,11 @@ def train_mnist_tune_checkpoint(
         logger=TensorBoardLogger(
             save_dir=tune.get_trial_dir(), name="", version="."),
         progress_bar_refresh_rate=0,
-        callbacks=[CheckpointCallback(),
-                   TuneReportCallback()])
+        callbacks=[CheckpointCallback("checkpoint", on="validation_end"),
+                   ReportCallback({
+                       "loss": "avg_val_loss",
+                       "mean_accuracy": "avg_val_accuracy"
+                   }, on="validation_end")])
```

Comment:
Wouldn't it make sense to put these two as 1 single callback?

CheckpointCallback wouldn't work if we didn't have a `ReportCallback`.

Reply:
I guess it still makes sense to have separate options (e.g. if someone doesn't want to use intermediate checkpointing for large models). There are probably three options: 1) Leave it separated as is, 2) Combine and make checkpointing optional, 3) Add a third, combined callback.
Right now I tend to 1 or 3, mostly because I think it's nice to be able to have checkpoints at different times than reports (e.g. after validation, but also after each training epoch). But it might not be needed in practice.

Reply:
I added a combined callback and used this in the tutorial. Internally it uses the other two callbacks. I guess the remaining question is if the CheckpointCallback on its own serves any purpose (after all, it does create checkpoints) or if we should mark it as internal by prepending a `_`.

Reply:
yeah i think it's safe to mark it as internal.